#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.6931032800836722 0.6931032800836722 0.1980295085953349 0.75
  position 1.2698955656624589 1.5260516968343767 2.309716969753342
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
DEF ROBOT Robot {
  translation 0 0.045 0
  rotation 0 0 -0.9999999999999978 0
  children [
    DEF EMITTER Emitter {
    }
    DEF RECEIVER Receiver {
    }
    DEF HINGE_COVER Solid {
      translation 0 0.03 -3.469446951953614e-18
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.6509803921568628 1
          }
          geometry Box {
            size 0.030000000000000002 0.019999999999999997 0.05
          }
        }
      ]
      name "hingeCover"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 6.723540149111039e-17
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 2.6326983297323222e-14 -2.2185872290083696e-14 1 1.570800000000675
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 6.670640318240083e-17
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 2.2263847525502005e-14 -1.9810582809503864e-14 1 1.570800000000661
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 1.0324509589061627e-16
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 3.272185563842641e-14 -3.391904931864592e-14 1 1.570800000000663
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position -2.6895122773572173e-17
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -5.384825698858949e-15 7.210054928651978e-15 1 1.5708000000006548
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF POLE HingeJoint {
      jointParameters HingeJointParameters {
        position -4.812689688184746e-16
        anchor 0 0.03000000000047226 0
        minStop -1.3
        maxStop 1.3
      }
      device [
        DEF POLE_POS_SENSOR PositionSensor {
          name "polePosSensor"
        }
      ]
      endPoint Solid {
        translation -1.8379558561219892e-06 0.529996887410479 -1.5902414965538783e-13
        rotation -1.5978770707143143e-13 1.5865798032821608e-13 1 1.570800003384083
        children [
          DEF POLE_ENDPOINT Solid {
            translation 0.5000000000000002 0 0
            scale 0.1 0.1 0.1
            children [
              DEF BALL Shape {
                appearance PBRAppearance {
                  baseColor 0.03137254901960784 0.047058823529411764 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.1
                }
              }
            ]
            boundingObject USE BALL
            physics Physics {
              density -1
              mass 0.02
            }
          }
          DEF POLE Shape {
            appearance PBRAppearance {
              baseColor 1 0.1568627450980392 0.1568627450980392
              roughness 1
              metalness 0
            }
            geometry Box {
              size 1 0.020000000000000004 0.019999999999999997
            }
          }
        ]
        name "pole"
        boundingObject USE POLE
        physics Physics {
          density -1
          mass 0.2
        }
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 2
  }
  controller "robotSupervisorController"
  supervisor TRUE
}
